### PR TITLE
JWWEB-631 Great directories structure changes

### PR DESCRIPTION
**Pull request by Creepypoke**
## What will do:

* Confirm/disconfirm the subdirectories for big modules like in `store/sagas/wallets`; ✅ 
* Update imports/exports from `store/sagas` and `store/modules` directories for more graceful and lapidary using; 🌗 
* ~Update imports in all files;~

## What is done:
1) All files from sagas and module directories were moved to store/sagas and store/module directories
2) Sagas exports were improved

### CODE REVIEW

File: src/routes/Wallets/routes/Import/WalletsImportView.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {
   WalletPasswordStep,
 } from 'components'
 
-import { STEPS } from './modules/walletsImport'
+import { STEPS } from '../../../../store/modules/walletsImport'
```
Comment (IViol) [inline src/routes/Wallets/routes/Import/WalletsImportView.js]: *just to mark big relative path*

File: src/store/modules/settings.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // @flow
 
-import type { CurrencyFormFieldValues } from '../routes/Currency/types'
+import type { CurrencyFormFieldValues } from '../../routes/Settings/routes/Currency/types'
```
Comment (IViol) [inline src/store/modules/settings.js]: *just to mark big relative path*

File: src/store/sagas.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
-import favoritesSagas from 'routes/Favorites/sagas'
-import digitalAssetsSagas from 'routes/DigitalAssets/sagas'
-import settingsSagas from 'routes/Settings/sagas'
+import coreSagas from 'store/sagas/index'
+import { upgradeRootSaga as upgradeSagas } from 'store/sagas/upgrade'
+import walletsSagas from 'store/sagas/wallets'
+import * as notFoundSagas from 'store/sagas/notFound'
+import { favoritesRootSaga as favoritesSagas } from 'store/sagas/favorites'
+import * as digitalAssetsSagas from 'store/sagas/digitalAssets/core'
+import { settingsRootSaga as settingsSagas } from 'store/sagas/settings'
```
Comment (IViol) [inline src/store/sagas.js]: I would like to left **clean** imports as is.
Combination of clean imports, `* as` and { fooRootSaga as fooSagas }` looks really bad.

File: src/store/sagas/digitalAssets/digitalAssetsSend.js [hunk 4]
```diff
@@ -52,9 +52,9 @@ import {
   selectDigitalAssetsSend,
 } from 'store/selectors/digitalAssets'
 
-import * as comments from 'routes/modules/comments'
-import * as transactions from 'routes/modules/transactions'
-import * as digitalAssetsSend from '../modules/digitalAssetsSend'
+import * as comments from 'store/modules/comments'
+import * as transactions from 'store/modules/transactions'
+import * as digitalAssetsSend from '../../modules/digitalAssetsSend'
```
Comment (IViol) [inline src/store/sagas/digitalAssets/digitalAssetsSend.js]: *just to mark big relative path*

File: src/store/sagas/digitalAssets/index.js [hunk 5]
```diff
@@ -2,23 +2,23 @@
...
 
 import {
   digitalAssetsGridRootSaga,
-} from '../routes/Grid/sagas/digitalAssetsGrid'
+} from './digitalAssetsGrid'
 
 import {
   digitalAssetsSendRootSaga,
-} from '../routes/Send/sagas/digitalAssetsSend'
+} from './digitalAssetsSend'
```
Comment (IViol) [inline src/store/sagas/digitalAssets/index.js]: I suggest such changes:
```
import { digitalAssetsCoreRootSaga } from './core'
import { digitalAssetsAddRootSaga } from './add'
import { digitalAssetsEditRootSaga } from './edit'
import { digitalAssetsGridRootSaga } from './grid'
import { digitalAssetsSendRootSaga } from './send'

export default {
  digitalAssetsCoreRootSaga,
  digitalAssetsAddRootSaga,
  digitalAssetsEditRootSaga,
  digitalAssetsGridRootSaga,
  digitalAssetsSendRootSaga,
}
```

File: src/store/sagas/wallets/index.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+import { walletsRootSaga as walletsSagas } from './core'
+import * as walletsAddressesSagas from './addresses'
+import * as walletsBackupSagas from './backup'
+import * as walletsCreateSagas from './create'
+import * as walletsDeleteSagas from './delete'
+import * as walletsImportSagas from './import'
+import * as walletsRenameSagas from './rename'
+import * as walletsRenameAddressSagas from './renameAddress'
+import * as walletsStarSagas from './start'
```
Comment (IViol) [inline src/store/sagas/wallets/index.js]: The same point about imports
Why we need to change existed notation - https://github.com/jibrelnetwork/jwallet-web/blob/master/src/routes/Wallets/sagas/index.js?

What is wrong with your version:
you do `import * as walletsCreateSagas from './create'`
but we need just one rootSaga in this file.

Let's discuss it offline

File: src/store/configureStore.js [hunk 7]
```diff
@@ -50,7 +51,11 @@ function configureStore(initialState: $Shape<AppState> = {}, history: Object) {
   // ======================================================
   // Run sagas
   // ======================================================
-  Object.keys(sagas).forEach(sagaName => sagaMiddleware.run(sagas[sagaName]))
+  forOwn(sagas, saga =>
+    isObject(saga) ?
+      forOwn(saga, sagaMiddleware.run)
+      : sagaMiddleware.run(saga)
+  )
```
Comment (IViol) [inline src/store/configureStore.js]: better to avoid unrelated changes

File: src/store/sagas/index.js [hunk 8]
```diff
@@ -1,17 +1,29 @@
...
 import { blocksRootSaga } from './blocks'
 import { tickerRootSaga } from './ticker'
 import { balancesRootSaga } from './balances'
 import { commentsRootSaga } from './comments'
 import { transactionsRootSaga } from './transactions'
+import { upgradeRootSaga } from './upgrade'
+import { favoritesRootSaga } from './favorites'
+import { settingsRootSaga } from './settings'
+import wallets from './wallets'
+import digitalAssets from './digitalAssets'
```
Comment (IViol) [inline src/store/sagas/index.js]: Let's name them `walletsSagas` & `digitalAssetsSagas`:
`import wallets from './wallets'` -> `import walletsSagas from './wallets'`

File: src/store/sagas/wallets/index.js (hunk 6 above)
Comment (IViol) [inline src/store/sagas/wallets/index.js]: walletsStarSagas -> walletsStartSagas

File: src/store/sagas/wallets/index.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+export {
+  getPrivateKey,
+  getPrivateKeyCancel,
+  GetPrivateKeyError,
+} from './core'
+
+export default {
+  ...{ walletsSagas },
```
Comment (IViol) [inline src/store/sagas/wallets/index.js]: `...{ walletsSagas }` -> `walletsSagas`

File: src/store/configureStore.js (hunk 7 above)
Comment (Creepypoke) [reply src/store/configureStore.js]: It's not unrelated changes, because sagas are exporting other way now

File: src/store/sagas/index.js (hunk 8 above)
Comment (Creepypoke) [reply src/store/sagas/index.js]: Good idea

File: src/store/sagas/digitalAssets/index.js (hunk 5 above)
Comment (Creepypoke) [reply src/store/sagas/digitalAssets/index.js]: This exports already in digitalAssets namespace, I guess the name like `coreRootSaga`, `editRootSaga` is enough
